scripts/top-search-queries/get-top-search-results.go

```diff
@@ -0,0 +1,242 @@
+
+	var responseObject searchResponse
+	responseObject.Items = searchItems
+	return responseObject
+}
+
+// AddResultsToCSV writes the results of a Search API query call to the supplied output CSV file (as separate rows).
+func AddResultsToCSV(ctx context.Context, querySupplied string, csvFile *os.File, nlpWeighting string, responseObject searchResponse) {
+	now := time.Now()
+	dateTimeRequest := now.Format(time.DateTime)
+	nlpOnOrOff := "off"
+	if nlpWeighting == "true" {
+		nlpOnOrOff = "on"
+	}
+	csvWriter := csv.NewWriter(csvFile)
+	defer csvWriter.Flush()
+
+	// The number of items in the responseObject will default to 10, but will be whatever is specified by the num_results input parameter, unless fewer are returned.
+	for i := 0; i < len(responseObject.Items); i++ {
+		item := responseObject.Items[i]
+		addItemToCSV(ctx, querySupplied, nlpOnOrOff, dateTimeRequest, item, i, csvWriter)
+	}
+}
+
+// addItemToCSV adds a row of data, taken from one item of the results (of a particular query), to the output CSV file.
+func addItemToCSV(ctx context.Context, querySupplied, nlpOnOrOff, dateTimeRequest string, item searchItem, position int, csvWriter *csv.Writer) {
+	rowNum := position + 1
+	resultsRow := []string{
+		nlpOnOrOff,
+		dateTimeRequest,
+		querySupplied,
+		item.ReleaseDate,
+		item.Type,
+		strconv.Itoa(rowNum),
+		item.Title,
+		item.Uri,
+		item.Edition,
+		item.Summary,
+	}
+
+	err := csvWriter.Write(resultsRow)
+	check(ctx, fmt.Sprintf("failed writing results row for query '%s' at row %d", querySupplied, rowNum), err)
+}
+
+// callSearchAPI calls the live Search API using the supplied values of query and nlpWeighting for the relevant query
+// parameters. It specifies a limit of 10 results to be returned in the query response.
+func callSearchAPI(ctx context.Context, query, nlpWeighting, apiURL string, numResults int) []models.Item {
+	queryVals := url.Values{}
+	queryVals.Add("q", query)
+	queryVals.Add("limit", strconv.Itoa(numResults))
+	queryVals.Add("nlp_weighting", nlpWeighting)
+	options := sdk.Options{
+		Query: queryVals,
+	}
+	logData := log.Data{"query values: ": queryVals}
+	log.Info(ctx, "using SDK to call Search API", logData)
+	searchAPIClient := sdk.New(apiURL)
+	response, err := searchAPIClient.GetSearch(ctx, options)
+	if err != nil {
+		log.Fatal(ctx, "failed calling Search API", err, logData)
+	}
+
+	return response.Items
+}
+
+// readQueriesFromFile opens the text file named "search-queries.txt", in the local directory, and reads it into a
+// []string object, which it returns.
+func readQueriesFromFile(ctx context.Context, cfg runConfig) (listOfQueries []string, err error) {
+	logData := log.Data{"File name: ": cfg.InputFileName}
+	file, err := os.Open(cfg.InputFileName)
+	if err != nil {
+		log.Fatal(ctx, "failed opening file", err, logData)
+	}
+	bufferedReader := bufio.NewReader(file)
+
+	var queryString string
+	for {
+		line, err := bufferedReader.ReadString('\n')
+		if err != nil {
+			if err == io.EOF {
```

Comment:
This function is unnecessarily complex and difficult to read, there are multiple lines where the query is being trimmed or appended. There is a much simpler way to read lines from files by using `buffio.Scanner` that overcomes this complexity.